digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/process/SpeedLimitValidator.scala

```diff
@@ -0,0 +1,114 @@
+package fi.liikennevirasto.digiroad2.process
+
+import fi.liikennevirasto.digiroad2.{GeometryUtils, Point}
+import fi.liikennevirasto.digiroad2.asset.{SideCode, TrafficDirection}
+import fi.liikennevirasto.digiroad2.dao.InaccurateAssetDAO
+import fi.liikennevirasto.digiroad2.dao.pointasset.PersistedTrafficSign
+import fi.liikennevirasto.digiroad2.linearasset.{NumericValue, RoadLink, SpeedLimit}
+import fi.liikennevirasto.digiroad2.service.pointasset.{TrafficSignService, TrafficSignType}
+import fi.liikennevirasto.digiroad2.service.pointasset.TrafficSignTypeGroup.SpeedLimits
+import fi.liikennevirasto.digiroad2.util.PolygonTools
+
+class SpeedLimitValidator(trafficSignService: TrafficSignService) {
+  val inaccurateAssetDAO: InaccurateAssetDAO = new InaccurateAssetDAO
+  val polygonTools: PolygonTools = new PolygonTools
+
+  private def minDistance(trafficSign: PersistedTrafficSign, point: Point) = Math.sqrt(Math.pow(trafficSign.lon - point.x, 2) + Math.pow(trafficSign.lat - point.y, 2))
+  private def min(s1: (Double, Seq[PersistedTrafficSign]), s2: (Double, Seq[PersistedTrafficSign])): (Double, Seq[PersistedTrafficSign]) = if (s1._1 < s2._1) s1 else s2
+
+  private def getTrafficSingsByRadius(speedLimit: SpeedLimit, roadLink: RoadLink): Seq[PersistedTrafficSign] = {
+    val speedLimitGeometry = GeometryUtils.truncateGeometry2D(roadLink.geometry, speedLimit.startMeasure, speedLimit.endMeasure)
+    val (first, last) = GeometryUtils.geometryEndpoints(speedLimitGeometry)
+
+    val trafficSingsByRadius =
+          (speedLimit.sideCode match {
+            case SideCode.TowardsDigitizing =>
+              trafficSignService.getTrafficSignByRadius(first, 50, Some(SpeedLimits)).filter (_.validityDirection == speedLimit.sideCode.value)
```

Comment:
Probably we could assign `trafficSignService.getTrafficSignByRadius(first, 50, Some(SpeedLimits))` to a variable, because it is the same operation in all the cases